bot/exts/utilities/bookmark.py

```diff
@@ -134,24 +186,41 @@ def event_check(reaction: discord.Reaction, user: discord.Member) -> bool:
                     user.id != self.bot.user.id
                 ))
             )
-        await self.action_bookmark(ctx.channel, ctx.author, target_message, title)
-
-        # Keep track of who has already bookmarked, so users can't spam reactions and cause loads of DMs
-        bookmarked_users = [ctx.author.id]
-        reaction_message = await self.send_reaction_embed(ctx.channel, target_message)
 
         while True:
             try:
-                _, user = await self.bot.wait_for("reaction_add", timeout=TIMEOUT, check=event_check)
+                reaction, user = await self.bot.wait_for("reaction_add", timeout=TIMEOUT, check=event_check)
             except asyncio.TimeoutError:
                 log.debug("Timed out waiting for a reaction")
                 break
             log.trace(f"{user} has successfully bookmarked from a reaction, attempting to DM them.")
             await self.action_bookmark(ctx.channel, user, target_message, title)
+            await reaction.remove()
             bookmarked_users.append(user.id)
 
         await reaction_message.delete()
 
+    @commands.dm_only()
+    @bookmark.command(name="delete", aliases=("del",), root_aliases=("unbm", "unbookmark"))
+    async def delete_bookmark(
+        self,
+        ctx: commands.Context,
+        message_to_delete: Optional[WrappedMessageConverter]
+    ) -> None:
+        """
+        Delete the referenced DM message by the user.
+
+        Users can either give a message as an argument, or reply to a message.
+        """
+        message_to_delete = message_to_delete or getattr(ctx.message.reference, "resolved", None)
+        if not message_to_delete:
+            raise commands.UserInputError(MESSAGE_NOT_FOUND_ERROR)
+
+        if message_to_delete.channel != ctx.channel:
```

Comment:
I'm not sure how it can be implemented, but it would be a nice addition to allow users to use the target bookmark message to delete. For example, they use `bm <message-id>` to bookmark, realized they bookmarked the wrong thing. So they go to their DMs and run `bm del <message-id>` to delete it. Here `<message-id>` would only need to be copied once and reused when deleting. Sure, they can just use the reply-to to delete, but making it like I said above would make it a sort of symmetric operation -- using the same argument to add and delete.

Reply:
Yea, that one would require use to store a lookup from message ID to dm message ID for each member, possibly over complicating it. I may be tempted to say we just keep the interface semi-restricted like this instead.

Reply:
I was going to say this wouldn't be easily possible since asyncredis cache doesn't really support these types, that's why I'm serialising. but then I realised if I'm already serialising, then I can just serialise dicts instead. I will rethink the approach for this.